File: api_gateway/src/models/product.py
from pydantic import BaseModel
from typing import Optional

class ProductModel(BaseModel):
  name:  str
  description: Optional[str]
  price: float
  quantity: int
  promotionPrice: float
  categoryId: str

class ProductFromSale(BaseModel):
  product_id: None | str = None
  price: float
  quantity: int

File: main/src/types/Product.ts
export type TProductCreated = {
  productId: string;
  name: string;
  description: string;
  price: number;
  promotionPrice: number;
  quantity: number;
  categoryId: string;
}

export type TCreationProductDTO = {
  name: string;
  description: string;
  price: number;
  promotionPrice: number;
  quantity: number;
  categoryId: string;
  storeId: string;
}